### ISSUE TITLE
Ignore node taints by default ( on GKE )

### ISSUE THREAD

**Issue by james-masson** [kubernetes-incubator/ip-masq-agent#11]
Hi,

Apologies if this isn't helpful, this issue is related to the configuration ip-masq-agent runs with under addon-manager in GKE. I'm having a tough time getting the GCP support guys to understand the problem, so I thought I'd post it here too, in the hope that the right people see it. (Case no 0016000000MB1EI/U-14664368 ) - If there's a better place to post this, please let me know.

My clusters run with a 172.16.x.x internal address range, and communicate with remote datacenters over a VPN link. If I don't have working ip-masq-agent (or equivalent), k8s cannot serve traffic to my remote datacenters.

The enforced config for ip-masq-agent on GKE is not to ignore node taints. This means that as soon as I use node taints, that node no longer runs ip-masq-agent, and may not be able to serve traffic correctly.

I can't think of a reason for this behavior - you either need ip-masq-agent running, or you don't. It should ignore node taints, like many other k8s critical services.

eg. Compare the tolerations for fluentd and ip-masq-agent on GKE

###  fluentd ###
      tolerations:
      - effect: NoSchedule
        key: node.alpha.kubernetes.io/ismaster
      - effect: NoExecute
        operator: Exists
      - effect: NoSchedule
        operator: Exists
###

### ip-masq-agent ###
      tolerations:
      - key: CriticalAddonsOnly
        operator: Exists
###

thanks

James M

**Comment by caseydavenport** [comment]
This seems sensible to me. We recently made the same change for Calico as well.

**Comment by pydevops** [comment]
@james-masson I found a new GKE (v1.8) 's ip-masq-agent doesn't with the custom configmap. still tried to understand it. On the other hand for your problem, you can look at https://blog.mrtrustor.net/post/iptables-kubernetes/ for a workaround running as a daemon set.

**Comment by james-masson** [comment]
@pydevops  - yes, that's is the workaround I'm trying to deprecate, now that ip-masq-agent on GKE supports configmap.

**Comment by james-masson** [comment]
this looks to be fixed now on GKE 1.8.8, thanks for fixing this, whoever you are!

**Comment by james-masson** [comment]
I spoke too soon. They've added more tolerations, but not enough to ignore a taint.
```
  tolerations:
  - key: CriticalAddonsOnly
    operator: Exists
  - effect: NoExecute
    key: node.alpha.kubernetes.io/notReady
    operator: Exists
  - effect: NoExecute
    key: node.alpha.kubernetes.io/unreachable
    operator: Exists
  - effect: NoSchedule
    key: node.kubernetes.io/disk-pressure
    operator: Exists
  - effect: NoSchedule
    key: node.kubernetes.io/memory-pressure
    operator: Exists
```

**Comment by kavehmz** [comment]
@james-masson I had the same issue. I filed a bug in google platform and issue is is fixed in 1.10.7.

https://github.com/kubernetes/kubernetes/pull/66260

I think we can close this ticket.

**Comment by MrHohn** [comment]
Closing per https://github.com/kubernetes-incubator/ip-masq-agent/issues/11#issuecomment-416135594.
